src/sentry/static/sentry/app/components/stackedBarChart.jsx

```diff
@@ -245,51 +260,80 @@ const StackedBarChart = createReactClass({
       }
     });
     return title;
-  },
+  };
 
-  renderChartColumn(point, maxval, pointWidth) {
+  getMinHeight(index, pointLength) {
+    let {minHeights} = this.props;
+    return minHeights && (minHeights[index] || minHeights[index] == 0)
+      ? this.props.minHeights[index]
+      : 1;
+  }
+
+  renderChartColumn(point, maxval, pointWidth, index, totalPoints) {
     let totalY = point.y.reduce((a, b) => a + b);
     let totalPct = totalY / maxval;
     let prevPct = 0;
     let pts = point.y.map((y, i) => {
-      let pct = totalY && this.floatFormat(y / totalY * totalPct * 99, 2);
+      let pct = Math.max(
+        totalY && this.floatFormat(y / totalY * totalPct * 99, 2),
+        this.getMinHeight(i, point.y.length)
+      );
+
       let pt = (
-        <span
+        <rect
           key={i}
-          className={this.props.barClasses[i]}
-          style={{
-            height: pct + '%',
-            bottom: prevPct + '%',
-            backgroundColor: this.state.series[i].color || null,
-          }}
+          x={index * pointWidth + '%'}
+          y={100.0 - pct - prevPct + '%'}
+          width={pointWidth + '%'}
+          height={pct + '%'}
+          fill={this.state.series[i].color}
+          className={cx(this.props.barClasses[i], 'barchart-rect')}
         >
           {y}
-        </span>
+        </rect>
       );
       prevPct += pct;
       return pt;
     });
 
     let pointIdx = point.x;
     let tooltipFunc = this.props.tooltip || this.renderTooltip;
+    let maskId = `path-${pointWidth}-${index}`; /* an ID in SVG is global just like html. */
 
     return (
       <Tooltip
         title={tooltipFunc(this.state.pointIndex[pointIdx], pointIdx, this)}
+        tooltipOptions={{html: true, placement: 'bottom', container: 'body'}}
         key={point.x}
-        tooltipOptions={{html: true, placement: 'bottom'}}
       >
-        <a className="chart-column" style={{width: pointWidth, height: '100%'}}>
-          {pts}
-        </a>
+        <g>
+          <rect
+            x={index * pointWidth + '%'}
+            width={pointWidth + '%'}
+            height="100%"
+            opacity="0"
+          />
+          <clipPath id={maskId}>
```

Comment:
We use a clip path and a transform to create a `calc` like effect (subtracting a fixed unit from the edge of each percent-based bar). If we don't care about having a fixed space between bars, then this can get a lot simpler, but I thought having the bars change width and the space between them remain the same was a nice side-effect of the old school float-based charts.